src/sql/platform/connection/test/common/connectionProfile.test.ts

```diff
@@ -236,7 +236,7 @@ suite('SQL ConnectionProfileInfo tests', () => {
 
 	test('getOptionsKey should create a valid unique id', () => {
 		let conn = new ConnectionProfile(capabilitiesService, iConnectionProfile);
-		let expectedId = 'providerName:MSSQL|connectionName:new name|databaseName:database|serverName:new server|userName:user|databaseDisplayName:database|groupId:group id';
+		let expectedId = 'providerName:MSSQL|authenticationType:|connectionName:new name|databaseName:database|encrypt:|serverName:new server|trustServerCertificate:|userName:user|databaseDisplayName:database|groupId:group id';
```

Comment:
Why are they included if they're empty? Is it because of missing STS side changes?

Reply:
Well, if authentication type is included because it is empty, then I decided to treat it the same here. Whatever is in STS should not affect the connection string here @cheenamalhotra

Reply:
Looks like I removed it in previous versions of the thing, so I will change it back

Reply:
I was mainly asking for encrypt and trustservercertificate - since they're advanced options and not set by user. So, we should either use defaults or not include at all.